scylla/src/transport/downgrading_consistency_retry_policy.rs

```diff
@@ -165,3 +165,466 @@ impl RetrySession for DowngradingConsistencyRetrySession {
+                );
+            }
+
+            // WriteType == UnloggedBatch
+            let write_type_unlogged_batch = QueryError::DbError(
+                DbError::WriteTimeout {
+                    consistency: LegacyConsistency::Regular(Consistency::Two),
+                    received,
+                    required,
+                    write_type: WriteType::UnloggedBatch,
+                },
+                String::new(),
+            );
+
+            for &cl in CONSISTENCY_LEVELS {
+                // Not idempotent
+                let mut policy = DowngradingConsistencyRetryPolicy::new().new_session();
+                assert_eq!(
+                    policy.decide_should_retry(make_query_info_with_cl(
+                        &write_type_unlogged_batch,
+                        false,
+                        cl
+                    )),
+                    RetryDecision::DontRetry
+                );
+
+                // Idempotent
+                let mut policy = DowngradingConsistencyRetryPolicy::new().new_session();
+                assert_eq!(
+                    policy.decide_should_retry(make_query_info_with_cl(
+                        &write_type_unlogged_batch,
+                        true,
+                        cl
+                    )),
+                    max_likely_to_work_cl(received, cl)
+                );
+                assert_eq!(
+                    policy.decide_should_retry(make_query_info_with_cl(
+                        &write_type_unlogged_batch,
+                        true,
+                        cl
+                    )),
+                    RetryDecision::DontRetry
+                );
+            }
+
+            // WriteType == other
+            let write_type_other = QueryError::DbError(
+                DbError::WriteTimeout {
+                    consistency: LegacyConsistency::Regular(Consistency::Two),
+                    received,
+                    required,
+                    write_type: WriteType::Simple,
+                },
+                String::new(),
+            );
+
+            for &cl in CONSISTENCY_LEVELS {
+                // Not idempotent
+                let mut policy = DowngradingConsistencyRetryPolicy::new().new_session();
+                assert_eq!(
+                    policy.decide_should_retry(make_query_info_with_cl(
+                        &write_type_other,
+                        false,
+                        cl
+                    )),
+                    RetryDecision::DontRetry
+                );
+
+                // Idempotent
+                let mut policy = DowngradingConsistencyRetryPolicy::new().new_session();
+                assert_eq!(
+                    policy.decide_should_retry(make_query_info_with_cl(
+                        &write_type_other,
+                        true,
+                        cl
+                    )),
+                    RetryDecision::IgnoreWriteError
+                );
```

Comment:
You should also check here that the policy returns `RetryDecision::DontRetry` afterwards.